src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -305,8 +305,37 @@ function importResourceTreeDirective($timeout, ImportContextService) {
             // Subscription handlers
             // =========================
 
-            const onResourcesUpdatedHandler = (resources) => {
-                $scope.resources = resources;
+            const onResourcesUpdatedHandler = (newResources = []) => {
+                const isUserImport = TABS.USER === ImportContextService.getActiveTabId();
+                if ($scope.resources.isEmpty()) {
+                    $scope.resources = ImportResourceTreeService.toImportResourceTree(_.cloneDeep(newResources), isUserImport);
+                    if (isUserImport) {
+                        $scope.sortedBy = SortingType.MODIFIED;
+                        sortResources();
+                    }
+                } else {
+                    // Removes missing files.
+                    $scope.resources.toList().forEach((resource) => {
```

Comment:
This can be moved in the resources model

Reply:
Nice. I will create an update method in the model and move all the update logic there.

Reply:
Done